pkg/kapp/cmd/app/delete.go

```diff
@@ -154,11 +155,8 @@ func (o *DeleteOptions) existingResources(app ctlapp.App,
 	fullyDeleteApp := true
 	applicableExistingResources := resourceFilter.Apply(existingResources)
 
-	if len(applicableExistingResources) != len(existingResources) {
-		fullyDeleteApp = false
-		o.ui.PrintLinef("App '%s' (namespace: %s) will not be fully deleted "+
-			"because some resources are excluded by filters",
-			app.Name(), o.AppFlags.NamespaceFlags.Name)
+	if !resourceFilter.IsEmpty() {
```

Comment:
This condition is important as `len(applicableExistingResources)` and `len(existingResources)` can be equal in two cases - 
1. On applying filter every resource is matched 
2. filter flag is not there then also we match all the resources 
So, to make sure we will mark `fullyDeleteApp ` flag as true only when user provided filter flag and everything matched.